plugins/c8y_configuration_plugin/src/download.rs

```diff
@@ -0,0 +1,299 @@
+use crate::error::ConfigDownloadError;
+use crate::smartrest::GetSmartRestMessage;
+use crate::{error, PluginConfig};
+use c8y_api::http_proxy::{C8YHttpProxy, JwtAuthHttpProxy};
+use c8y_smartrest::error::SmartRestSerializerError;
+use c8y_smartrest::smartrest_deserializer::SmartRestConfigDownloadRequest;
+use c8y_smartrest::smartrest_serializer::{
+    CumulocitySupportedOperations, SmartRest, SmartRestSerializer,
+    SmartRestSetOperationToExecuting, SmartRestSetOperationToFailed,
+    SmartRestSetOperationToSuccessful,
+};
+use download::{Auth, DownloadInfo, Downloader};
+use mqtt_channel::{Connection, Message, SinkExt, Topic};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::fs;
+use std::os::unix::fs::PermissionsExt;
+use std::path::{Path, PathBuf};
+use tedge_config::{ConfigSettingAccessor, TEdgeConfig, TmpPathDefaultSetting};
+
+const BROADCASTING_TOPIC: &str = "filemanagement/changes";
```

Comment:
I'm not convinced by this topic name. I would prefer something along the lines `tedge/c8y_configuration`.

Reply:
I used the Albin's proposal.